File: backend/app/Models/Employee.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Employee extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'institution_id',
        'branch_id',
        'department_id',
        'position_id',
        'employee_number',
        'full_name',
        'email',
        'phone',
        'birth_date',
        'address',
        'join_date',
        'status',
    ];

    protected $casts = [
        'birth_date' => 'date',
        'join_date' => 'date',
    ];

    // Relationships
    public function institution()
    {
        return $this->belongsTo(Institution::class);
    }

    public function branch()
    {
        return $this->belongsTo(Branch::class);
    }

    public function department()
    {
        return $this->belongsTo(Department::class);
    }

    public function position()
    {
        return $this->belongsTo(Position::class);
    }

    public function user()
    {
        return $this->hasOne(User::class);
    }

    public function attendances()
    {
        return $this->hasMany(EmployeeAttendance::class);
    }

    public function leaveRequests()
    {
        return $this->hasMany(LeaveRequest::class);
    }

    public function leaveBalances()
    {
        return $this->hasMany(EmployeeLeaveBalance::class);
    }

    public function overtimeRequests()
    {
        return $this->hasMany(OvertimeRequest::class);
    }

    public function payslips()
    {
        return $this->hasMany(Payslip::class);
    }

    public function schedules()
    {
        return $this->hasMany(DailySchedule::class);
    }
}

File: mobile/lib/domain/entities/employee.dart
import 'package:equatable/equatable.dart';

class Employee extends Equatable {
  final int id;
  final String employeeNumber;
  final String fullName;
  final String? email;
  final String? phone;
  final String? address;
  final DateTime? dateOfBirth;
  final String? gender;
  final int? departmentId;
  final String? departmentName;
  final int? positionId;
  final String? positionName;
  final int? branchId;
  final String? branchName;
  final DateTime? joinDate;
  final String? status;

  const Employee({
    required this.id,
    required this.employeeNumber,
    required this.fullName,
    this.email,
    this.phone,
    this.address,
    this.dateOfBirth,
    this.gender,
    this.departmentId,
    this.departmentName,
    this.positionId,
    this.positionName,
    this.branchId,
    this.branchName,
    this.joinDate,
    this.status,
  });

  @override
  List<Object?> get props => [
        id,
        employeeNumber,
        fullName,
        email,
        phone,
        address,
        dateOfBirth,
        gender,
        departmentId,
        departmentName,
        positionId,
        positionName,
        branchId,
        branchName,
        joinDate,
        status,
      ];
}
